core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexAliasesService.java

```diff
@@ -56,96 +62,95 @@
     private final AliasValidator aliasValidator;
 
     private final NodeServicesProvider nodeServicesProvider;
+    
+    private final MetaDataDeleteIndexService deleteIndexService;
 
     @Inject
-    public MetaDataIndexAliasesService(Settings settings, ClusterService clusterService, IndicesService indicesService, AliasValidator aliasValidator, NodeServicesProvider nodeServicesProvider) {
+    public MetaDataIndexAliasesService(Settings settings, ClusterService clusterService, IndicesService indicesService,
+            AliasValidator aliasValidator, NodeServicesProvider nodeServicesProvider, MetaDataDeleteIndexService deleteIndexService) {
         super(settings);
         this.clusterService = clusterService;
         this.indicesService = indicesService;
         this.aliasValidator = aliasValidator;
         this.nodeServicesProvider = nodeServicesProvider;
+        this.deleteIndexService = deleteIndexService;
     }
 
-    public void indicesAliases(final IndicesAliasesClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
-        clusterService.submitStateUpdateTask("index-aliases", new AckedClusterStateUpdateTask<ClusterStateUpdateResponse>(Priority.URGENT, request, listener) {
+    public void indicesAliases(final IndicesAliasesClusterStateUpdateRequest request,
+            final ActionListener<ClusterStateUpdateResponse> listener) {
+        clusterService.submitStateUpdateTask("index-aliases",
+                new AckedClusterStateUpdateTask<ClusterStateUpdateResponse>(Priority.URGENT, request, listener) {
             @Override
             protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
                 return new ClusterStateUpdateResponse(acknowledged);
             }
 
             @Override
-            public ClusterState execute(final ClusterState currentState) {
+            public ClusterState execute(ClusterState currentState) {
```

Comment:
Would be cool if we can add a unit test for this method if it doesn't take too much effort. We can move the core logic to a package protected method called something like `innerExecute(...)` and call that from this method and the unit test. For certain fields like  `metaDataDeleteIndexService` we can use mocks. This should be work out?